aopy/analysis/base.py

```diff
@@ -2372,4 +2376,261 @@ def simulate_ideal_trajectories(targets, origin=[0.0, 0.0, 0.0], resolution=1000
+    if callable(statistics):
+        statistics=[statistics] * len(conditions)
+
+    cond_dists = []
+    for idx, cond in enumerate(conditions):
+        try:
+            data_cond = data[labels == cond]
+        except KeyError:
+            data_cond = data.iloc[labels == cond].reset_index(drop=True)
+
+        if len(data_cond) < n_trials:
+            raise ValueError(f"Condition {cond} has less than {n_trials} trials")
+
+        dist = calc_trial_bootstraps(data_cond, n_trials=n_trials, n_bootstraps=n_bootstraps,
+                                     statistic=statistics[idx], parallel=parallel)
+        
+        cond_dists.append(dist)
+
+    # Compare spatial correlation
+    corr_mats = []
+    for i in range(n_bootstraps):
+        dists = [cd[i] for cd in cond_dists]
+        cmp = calc_spatial_data_correlation(np.concatenate(dists), 
+                                            elec_pos, interp=False)[0]
+        corr_mats.append(cmp)
+
+    # d-prime pooled across bootstraps
+    dprime = tuning.calc_dprime(*[np.concatenate(cd) for cd in cond_dists])
+
+    return cond_dists, conditions, corr_mats, dprime
+
+def compare_conditions_bootstrap_spatial_corr(
+        data, elec_pos, labels, n_trials=300, n_bootstraps=30, n_shuffle=0, 
+        statistics=partial(np.mean, axis=0), parallel=False):
+    '''
+    Compare multiple conditions using bootstrapping and shuffling. 
+
+    Args:
+        data (ntrials, nch): data to calculate the statistic on
+        elec_pos ((nch, 2) array): electrode positions for each channel
+        labels (ntrials,): labels for each trial, used to group trials by condition
+        n_trials (int): number of trials to use in each bootstrap
+        n_bootstraps (int): number of bootstrap trials to perform
+        n_shuffle (int): number of times to shuffle the labels to create a null distribution
+        statistics (function or list of functions): function(s) to calculate the statistic on the data. If more than one is supplied, they will be applied per condition 
+
+    Returns:
+        tuple: tuple containing:
+            | **observed_dists (n_cond, n_bootstraps, len(data)//n_trials, nch):** bootstraps distributions for each condition
+            | **conditions (n_cond):** unique conditions in the labels
+            | **observed_corr (n_bootstraps, n_cond*n_cond, n_cond*n_cond, nch):** list of spatial correlation matrices for each bootstrap
+            | **observed_dprime (nch):** d-prime value for each channel pooled across bootstraps and comparing across conditions
+        
+        if n_shuffle > 0:
+            | **shuff_dists_dist (n_shuffle, n_cond, n_bootstraps, len(data)//n_trials, nch):** shuffled distributions for each condition
+            | **shuff_corr_dist (n_shuffle, n_bootstraps, n_cond*n_cond, n_cond*n_cond, nch):** shuffled spatial correlation matrices
+            | **shuff_dprime_dist (n_shuffle, nch):** shuffled d-prime   
```

Comment:
if this can be 0, why not just have shuffling as part of the previous function instead of 2 separate functions?